### PR TITLE
Remove the JSDependencies plugin from our own build.

### PR DESCRIPTION
**Pull request by sjrd**
This is of course required before we can actually move that plugin in a separate repo, and remove from the core repo.

### CODE REVIEW

File: project/Build.scala [hunk 1]
```diff
@@ -638,6 +638,12 @@ object Build {
         testDefinitionsFile +: (jsExecutionFiles in Test).value
       },
 
+      jsExecutionFiles in Test := {
+        val resourceDir = (resourceDirectory in (testSuite, Test)).value
+        val f = FileVirtualJSFile(resourceDir / "ScalaJSDefinedTestNatives.js")
+        f +: (jsExecutionFiles in Test).value
+      },
```
Comment (gzm0) [inline project/Build.scala]: Have you considered factoring this into a method (that takes a key)?

Comment (sjrd) [inline project/Build.scala]: I, er, haven't. You mean something like this, I guess:
```scala
def testJSExecutionFilesSetting: Setting[_] = {
  jsExecutionFiles in Test := {
    val resourceDir = (resourceDirectory in (testSuite, Test)).value
    val f = FileVirtualJSFile(resourceDir / "ScalaJSDefinedTestNatives.js")
    f +: (jsExecutionFiles in Test).value
  }
}
```
and then use it both in `testSuite.settings(...)` and `toolsJS.settings(...)`?

Comment (gzm0) [inline project/Build.scala]: Yeah, something the like :)

Comment (sjrd) [inline project/Build.scala]: OK. Will wait for #2975 to go through before rebasing.

Comment (sjrd) [inline project/Build.scala]: Done.

**Pull request merged**